mlflow/pyfunc/__init__.py

```diff
@@ -832,11 +838,20 @@ def predict(*args):
             result = result.select_dtypes(include=(np.number,)).astype(np.float64)
 
         if len(result.columns) == 0:
+            message = (
+                "The model did not produce any values compatible with the requested type "
+                "'{}'. ".format(str(elem_type))
+            )
+            if len(args) == 0 and input_schema is None:
+                message += (
+                    "Apply the udf by specifying column name arguments, for example "
```

Comment:
I meant to add this message at the top level. If that is not possible, I would not add it at all because I don't think we can add the message reliably  at the python level - by which I mean some models will raise exceptions, some will produce empty output, it would be difficult to reliably detect when predict fails due to missing columns.